pkg/planner/core/plan_cache.go

```diff
@@ -796,36 +787,6 @@ func CheckPreparedPriv(sctx sessionctx.Context, stmt *PlanCacheStmt, is infosche
 	return err
 }
 
-// tryCachePointPlan will try to cache point execution plan, there may be some
-// short paths for these executions, currently "point select" and "point update"
-func tryCachePointPlan(_ context.Context, sctx base.PlanContext,
-	stmt *PlanCacheStmt, p base.Plan, names types.NameSlice) error {
-	if !sctx.GetSessionVars().StmtCtx.UseCache() {
-		return nil
-	}
-	var (
-		ok  bool
-		err error
-	)
-
-	if plan, _ok := p.(*PointGetPlan); _ok {
-		ok = IsPointGetWithPKOrUniqueKeyByAutoCommit(sctx.GetSessionVars(), p)
```

Comment:
In the new code logic, don't we need to check "IsPointGetWithPKOrUniqueKeyByAutoCommit" ?

Reply:
No, this check is unnecessary for Plan Cache, we only need to check whether the Plan is *PointGetPlan.